service/create/service.go

```diff
@@ -1383,3 +1418,24 @@ func (s *Service) updateFunc(oldObj, newObj interface{}) {
 
 	s.logger.Log("info", fmt.Sprintf("processed autoscaling group stack '%s'", key.AutoScalingGroupName(cluster, prefixWorker)))
 }
+
+func (s *Service) findCFASG(client *autoscaling.AutoScaling, name string, clusterID string) (awsresources.AutoScalingGroup, error) {
```

Comment:
This API call should really be in the `resources/aws` package. I'm OK with it since its migration code. If we need it long term it should be moved.

Reply:
IMO there is lots of room for improvement and the code structure will change anyway as soon as we move to the reconciler framework.

Reply:
Exactly, I didn't spend too much time organising the code given the upcoming changes.

Reply:
Cool location is fine since lots of code restructuring is coming.

Reply:
Thank you Ross, you are completely right about this code belonging to `resources/aws`, probably there are things also in the ASG resource that don't belong there either, let's see how we can structure our logic in the next steps.